staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -167,8 +131,8 @@ func WithAuditContext(parent context.Context) context.Context {
 
```

Comment:
I think it's safer for the the initial context to return a `LevelNone` for the `Level` field

```
func WithAuditContext(parent context.Context) context.Context {
	if AuditContextFrom(parent) != nil {
		return parent // Avoid double registering.
	}
        ac := &AuditContext{
              Event: {
                  Level: LevelNone
              }
        }
	return genericapirequest.WithValue(parent, auditKey, ac)
}
```

https://github.com/kubernetes/kubernetes/blob/17bf864c1fc20e4badc9a6e0659c5190310462e1/staging/src/k8s.io/apiserver/pkg/audit/context.go#L159-L166

Reply:
See above comment about capturing data before the policy is evaluated.